compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -169,7 +169,16 @@ impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for ExpnId {
 impl<'a, 'tcx> Encodable<EncodeContext<'a, 'tcx>> for Span {
     fn encode(&self, s: &mut EncodeContext<'a, 'tcx>) {
         match s.span_shorthands.entry(*self) {
-            Entry::Occupied(o) => SpanEncodingMode::Shorthand(*o.get()).encode(s),
+            Entry::Occupied(o) => {
+                // If an offset is smaller than the absolute position, we encode with the offset.
+                // This saves space since smaller numbers encode in less bits.
+                let offset = s.opaque.position() - *o.get();
```

Comment:
can this ever underflow? (if it can the code probably needs to account for that, if it can't, it would be nice to have a comment)

Reply:
No, it shouldn't be possible. We write ~all metadata bytes strictly increasing positions in the buffer, so the current position must be larger than the position we saved into the HashMap.

I'll add a comment.